Composer/packages/form-dialogs/src/undo/UndoRoot.tsx

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
```

Comment:
This is a significant amount of work to support undo.  Maybe we can lift this up to shared across Composer for other plug-ins and extensions.

Reply:
It's pretty much this one file.